app/controllers/api/v3/notices_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Api::V3::NoticesController < ApplicationController
+
+  UNKNOWN_API_KEY = "Your API key is unknown".freeze
+
+  skip_before_action :verify_authenticity_token
+  skip_before_action :authenticate_user!
+
+  respond_to :json
+
+  def create
+    response.headers["Access-Control-Allow-Origin"] = "*"
+    response.headers["Access-Control-Allow-Headers"] = "origin, content-type, accept"
+    return render(status: 200, text: "") if request.method == "OPTIONS"
+
+    merged_params = params.merge!(!request.raw_post.empty? && JSON.parse(request.raw_post) || {})
+    # merge makes a copy, merge! edits in place
+    merged_params = merged_params.merge!("key" => request.headers["X-Airbrake-Token"]) if request.headers["X-Airbrake-Token"]
+    merged_params = merged_params.merge!("key" => authorization_token) if authorization_token
```

Comment:
Sorry, just noticed this...

👆 This is weird! First, `params.merge!` modifies `params`, so `merged_params` and `params` are identical. Second, there's no need to assign the result of `merged_params.merge!` to `merged_params`.

I think the original coder of this might not have had a great understanding this method in Ruby.

This might be the intent:

```ruby
merged_params = params.merge(!request.raw_post.empty? && JSON.parse(request.raw_post) || {})
merged_params.merge!("key" => request.headers["X-Airbrake-Token"]) if request.headers["X-Airbrake-Token"]
merged_params.merge!("key" => authorization_token) if authorization_token
```